### ISSUE TITLE
Wrapping SMDS_ElemIteratorPtr and SMDS_NodeIteratorPtr

### ISSUE THREAD

**Issue by trelau** [tpaviot/pythonocc-core#404]
SMDS_ElemIteratorPtr and SMDS_NodeIteratorPtr appear the way you access and iterate through nodes and elements within a mesh. When I try and access a method that returns one of these types, I get the following message:

![image](https://cloud.githubusercontent.com/assets/8950602/23473943/0e96d262-fe80-11e6-9d61-610f2e413947.png)

I tried a few things based on the info [here ](http://swig.org/Doc3.0/Library.html#Library_std_shared_ptr) and [here](http://stackoverflow.com/questions/18878179/how-to-avoid-memory-leak-with-shared-ptr-and-swig), which at this point is simply adding this to the SMDS.i file:

```
%include "boost_shared_ptr.i"
%shared_ptr(SMDS_ElemIteratorPtr)
%shared_ptr(SMDS_NodeIteratorPtr)
```

and got the same results. I'll keep working it, but it seems like a necessary functionality to be able to work with meshes.

Here is the entities that I think need wrapped: https://github.com/tpaviot/smesh/blob/review/fc-smesh-771/inc/SMDS_ElemIterator.hxx#L42

[Here](https://github.com/trelau/pythonocc-core/commit/2dc6797aec7110d259564dcf3d343a7bed3ac1ce) is a first attempt to make these iterators available by manually modifying the SMDS.i file. This makes the iterators available and the more(), next(), remove() now seem wrapped but more() causes a fatal crash and next() returns None...

**Update**
It seems I have it working. See commit [here](https://github.com/trelau/pythonocc-core/commit/edacd26d816afa669fd8b9c0174a65c8ffb203c8). In summary, we have to include and define a boost shared_ptr, wrap the SMDS_Iterator class, and then define the Ptr templates. After that you can access the IteratorPtr and they return SMDS_MeshNode/SMDS_MeshElement.

![image](https://cloud.githubusercontent.com/assets/8950602/23512803/e6d5e0f0-ff2f-11e6-9ef0-987779f45dc6.png)

![image](https://cloud.githubusercontent.com/assets/8950602/23512812/ef67e1d2-ff2f-11e6-9c18-b880ca6b4205.png)

**Comment by tpaviot** [comment]
@trelau seems to work, thanks see 84d1968a4677adaf86f6ab401cde5f50a979542c

**Comment by tpaviot** [comment]
Fixed by commit 5ca2062c9a127f6ab74f2629e0819798b8f0d1d0
